mods/persistence/modules/trade_beacons/exports.dm

```diff
@@ -0,0 +1,80 @@
+
+/datum/beacon_export
+	var/name = "Beacon Export"
+	var/list/required_items = list() // structure is list(/obj/item/example = 1, /obj/item/multi = 2) list(type = number required).
+	var/cost = 1000
+	var/remaining_stock = 5
+	var/desc = "Export Description"
+	var/obj/checked_closet
+/datum/beacon_export/proc/get_cost(var/tax)
+	if(!tax) return cost
+	return cost-((tax/100)*cost)
+
+/datum/beacon_export/proc/get_tax(var/tax)
+	if(!tax) return 0
+	return (tax/100)*cost
+
+/datum/beacon_export/proc/takeExport()
+	if(checked_closet)
+		checked_closet.forceMove(null)
+		qdel(checked_closet)
+	checked_closet = null
+	remaining_stock--
+/datum/beacon_export/proc/resetExport()
+	checked_closet = null
+
+/datum/beacon_export/proc/checkExport(var/turf/L)
+	if(!L) return "No valid location detected"
+	var/obj/closet_found = 0
+	var/valid_items = list()
+	for(var/obj/structure/closet/C in L.contents)
```

Comment:
This loop is very crowded. Could you split it into several procs? After 2-3 indents it becomes a bit much and hard to read. And it's easier to have an indent bug that's super hard to track down after.

Reply:
I've looked into splitting up into procs but it would still be very messy, and the existing proc is small enough to fit on one screen. I agree with what you're saying about excessive indents but splitting it into procs would mean passing around lists of valid and invalid items. This particular block is for checking the contents of exports before they are taken and I've tested it in game.